### PR TITLE
Use globalArgs for theming addon-docs

### PR DESCRIPTION
**Pull request by jpzwarte**
Issue:

## What I did
Replaced the theming code in `DocsContainer.tsx` with globalArgs.

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by shilman** [comment]
cc @ndelangen @tmeasday WDYT?

**Comment by tmeasday** [comment]
@shilman I am guessing you wouldn't be able to do complicated things here because globalArgs needs to be serialized over the channel etc etc.

It would be kind of cool if Storybook's light/dark theming was controllable via globalArgs though.

**Comment by shilman** [comment]
@tmeasday the theme vars are serializable over the channel AFAIK. the issue is that manager has `setConfig` to update the global theme but Docs theme runs in the preview. it currently uses story parameters to set the theme, but story parameters shouldn't be modified at runtime. this could get around that.

**Comment by jpzwarte** [comment]
What about having a single way to theme storybook? So you use globalArgs for the core theming as well?

So this would introduce 2 config settings:
```
export const globalArgs = {
  colorScheme: 'light|dark',
  themes: {
    light: ThemeVars,
    dark: ThemeVars
  }
};
```

`colorScheme` naming matches the media query used for light/dark: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme.

All are optional, which results in a light default theme, but one could provide their own themes and change (runtime!) the color scheme in a Tool (similar to `storybook-dark-mode`). 

This would fix #10523.

**Comment by jpzwarte** [comment]
@shilman afaik from trying, you can only use `setConfig` during startup to set the theme. At runtime, you have to use `api.setOptions` to change the theme.

**Comment by tmeasday** [comment]
Seems like a reasonable approach for sure.

We can add this as non-breaking change in 6.1 so I think we should probably hold off given we are days away from 6.0rc.

File: addons/docs/src/blocks/DocsContainer.tsx [hunk 1]
```diff
@@ -3,9 +3,10 @@ import { document, window } from 'global';
 import deprecate from 'util-deprecate';
 import dedent from 'ts-dedent';
 import { MDXProvider } from '@mdx-js/react';
-import { ThemeProvider, ensure as ensureTheme } from '@storybook/theming';
+import { ThemeProvider, ensure as ensureTheme, themes as builtInThemes } from '@storybook/theming';
 import { DocsWrapper, DocsContent } from '@storybook/components';
 import { components as htmlComponents } from '@storybook/components/html';
+import { useGlobalArgs } from '@storybook/api';
```
Comment (lopis) [inline addons/docs/src/blocks/DocsContainer.tsx:9]: Not sure I'm missing something, but where is this import being used?

**Comment by shilman** [comment]
FYI, breaking prerelease change https://github.com/storybookjs/storybook/pull/11385

**Comment by ndelangen** [comment]
A great step to take would be to move the definition of themes from `manager.js` to main.js:

```js
module.exports = {
  theme: light,
};
```

**Comment by ndelangen** [comment]
This is currently unmergable because of the merge conflicts, I think.. 

Can we rebase this?

**Comment by jpzwarte** [comment]
@ndelangen Feel free to use my code as you see fit; unfortunately I don't have time to work on this PR myself atm.

**Pull request closed without merging**